keps/sig-node/2371-cri-pod-container-stats/README.md

```diff
@@ -0,0 +1,416 @@
+-message MemoryUsage {
+-    // Timestamp in nanoseconds at which the information were collected. Must be > 0.
++    // Corresponds to Stats Summary API MemoryStats Time field
+-    int64 timestamp = 1;
+-    // The amount of working set memory in bytes.
++    // Corresponds to Stats Summary API MemoryStats WorkingSetBytes field
+-    UInt64Value working_set_bytes = 2;
++    // Available memory for use.  This is defined as the memory limit - workingSetBytes.
++    // If memory limit is undefined, the available bytes is omitted.
++    UInt64Value available_bytes = 3;
++    // Total memory in use. This includes all memory regardless of when it was accessed.
++    UInt64Value usage_bytes
++    // The amount of working set memory. This includes recently accessed memory,
++    // dirty memory, and kernel memory. WorkingSetBytes is <= UsageBytes
++    UInt64Value working_set_bytes = 4;
++    // The amount of anonymous and swap cache memory (includes transparent
++    // hugepages).
++    UInt64Value rss_bytes = 5;
++    // Cumulative number of minor page faults.
++    Uint64Value page_faults = 6;
++    // Cumulative number of major page faults.
++    Uint64Value major_page_faults = 6;
+-}
+
+-// FilesystemUsage provides the filesystem usage information.
+-message FilesystemUsage {
+-    // Timestamp in nanoseconds at which the information were collected. Must be > 0.
++    // Corresponds to Stats Summary API FsStats Time field
+-    int64 timestamp = 1;
+-    // The unique identifier of the filesystem.
++    // Does not correspond to any field in Stats Summary API FsStats
+-    FilesystemIdentifier fs_id = 2;
+-    // UsedBytes represents the bytes used for images on the filesystem.
+-    // This may differ from the total bytes used on the filesystem and may not
+-    // equal CapacityBytes - AvailableBytes.
+-    // Corresponds to Stats Summary API FsStats UsedBytes field
+-    UInt64Value used_bytes = 3;
+-    // InodesUsed represents the inodes used by the images.
+-    // This may not equal InodesCapacity - InodesAvailable because the underlying
+-    // filesystem may also be used for purposes other than storing images.
+-    // Corresponds to Stats Summary API FsStats InodesUsed field
+-    UInt64Value inodes_used = 4;
++    // TODO: Unclear how the remaining fields relate to container stats. Is it filled in cAdvisor?
++    // AvailableBytes represents the storage space available (bytes) for the filesystem.
++    UInt64Value available_bytes = 5;
++    // CapacityBytes represents the total capacity (bytes) of the filesystems underlying storage.
++    UInt64Value capacity_bytes = 6;
++    // InodesFree represents the free inodes in the filesystem.
++    UInt64Value inodes_free = 7;
++    // Inodes represents the total inodes in the filesystem.
++    UInt64Value inodes = 8;
+-}
+```
+
+**TODO**: `UserDefinedMetrics` aren't supported by CRI and are cAdvisor specific, does it make sense to even support them in new CRI interfaces?
```

Comment:
This TODO appears to be resolved above, but in-case it isn't... I would not support UserDefinedMetrics

Reply:
I have dropped the TODO, neither @bobbypage nor I want to support them, so your recommendation is the nail in the coffin imo :)